src/databricks/sql/cloudfetch/downloader.py

```diff
@@ -19,85 +24,125 @@ def __init__(self, downloadable_result_settings, t_spark_arrow_result_link):
         self.is_file_downloaded_successfully = False
         self.is_link_expired = False
         self.is_download_timedout = False
-        self.http_code = None
         self.result_file = None
-        self.check_result_file_link_expiry = True
-        self.download_completion_semaphore = threading.Semaphore(0)
 
-    def is_file_download_successful(self):
+    def is_file_download_successful(self) -> bool:
+        """
+        Check and report if cloud fetch file downloaded successfully.
+
+        This function will block until a file download finishes or until a timeout.
+        """
+        timeout = self.settings.download_timeout
+        timeout = timeout if timeout and timeout > 0 else None
         try:
-            if not self.is_download_finished.is_set():
-                if self.settings.download_timeout and self.settings.download_timeout > 0:
-                    if not self.download_completion_semaphore.acquire(timeout=self.settings.download_timeout):
-                        self.is_download_timedout = True
-                        logger.debug("Cloud fetch download timed out after {} seconds for url: {}"
-                                     .format(self.settings.download_timeout, self.result_link.file_link)
-                                     )
-                        return False
-                else:
-                    self.download_completion_semaphore.acquire()
-        except:
+            if not self.is_download_finished.wait(timeout=timeout):
+                self.is_download_timedout = True
+                logger.debug(
+                    "Cloud fetch download timed out after {} seconds for url: {}".format(
+                        self.settings.download_timeout,
+                        self.result_link.fileLink,
```

Comment:
I don't think we should be logging the actual URLs

Reply:
Oh I guess even with `debug` that may be a hazard. Will change